tests/test_classify_pagination.py

```
@@ -257,6 +258,189 @@ def test_v2_checkpoint_with_timestamp_preserved(self):
             os.unlink(temp_path)
 
 
+class TestProjectTagClassificationSource(unittest.TestCase):
+    """Tests classification source selection and discovery reset behavior."""
+
+    def setUp(self):
+        self.temp_db = tempfile.NamedTemporaryFile(suffix='.db', delete=False)
```

Ruff (0.15.15)

[warning] Use a context manager for opening files

(SIM115)